src/plugins/ui_actions/public/cell_actions/components/hover_actions_popover.tsx

```diff
@@ -42,127 +42,138 @@ interface Props {
+    };
+  }, [openPopOverDebounced]);
+
+  const onMouseEnter = useCallback(async () => {
+    // Do not open actions with extra action popover is open
+    if (isExtraActionsPopoverOpen) return;
+
+    // memoize actions after the first call
+    if (actions === undefined) {
+      loadActions(actionContext);
+    }
+
+    openPopOverDebounced();
+  }, [isExtraActionsPopoverOpen, actions, openPopOverDebounced, loadActions, actionContext]);
+
+  const onMouseLeave = useCallback(() => {
+    closePopover();
+  }, [closePopover]);
+
+  const content = useMemo(() => {
     return (
-      <>
-        <div onMouseLeave={onMouseLeave}>
-          <EuiPopover
-            panelStyle={PANEL_STYLE}
-            ref={popoverRef}
-            anchorPosition={'downCenter'}
-            button={content}
-            closePopover={closePopover}
-            hasArrow={false}
-            isOpen={showHoverContent}
-            panelPaddingSize="none"
-            repositionOnScroll
-            ownFocus={false}
-            data-test-subj={'hoverActionsPopover'}
-            aria-label={ACTIONS_AREA_LABEL}
-          >
-            {showHoverContent ? (
-              <div css={hoverContentWrapperCSS}>
-                <EuiScreenReaderOnly>
-                  <p>{YOU_ARE_IN_A_DIALOG_CONTAINING_OPTIONS(actionContext.field.name)}</p>
-                </EuiScreenReaderOnly>
-                {visibleActions.map((action) => (
-                  <ActionItem
-                    key={action.id}
-                    action={action}
-                    actionContext={actionContext}
-                    showTooltip={showActionTooltips}
-                  />
-                ))}
-                {extraActions.length > 0 ? (
-                  <ExtraActionsButton
-                    onClick={onShowExtraActionsClick}
-                    showTooltip={showActionTooltips}
-                  />
-                ) : null}
-              </div>
-            ) : null}
-          </EuiPopover>
-        </div>
-        <ExtraActionsPopOverWithAnchor
-          actions={extraActions}
-          anchorRef={contentRef}
-          actionContext={actionContext}
-          closePopOver={closeExtraActions}
-          isOpen={isExtraActionsPopoverOpen}
-        />
-      </>
+      // Hack - Forces extra actions popover to close when hover content is clicked.
+      // This hack is required because we anchor the popover to the hover content instead
+      // of anchoring it to the button that triggers the popover.
+      // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+      <div ref={contentRef} onMouseEnter={onMouseEnter} onClick={closeExtraActions}>
+        {children}
+      </div>
     );
-  }
-);
+  }, [onMouseEnter, closeExtraActions, children]);
+
+  useEffect(() => {
+    if (error) {
+      // eslint-disable-next-line no-console -- Make sure loadActions error are not swallowed
+      console.error(error);
+    }
+  }, [error]);
```

Comment:
I don't think we should do `console.error` to treat errors. 
I think the best we could do is just `throw` the error up, because this is an isolated component, we don't have access to toasts or anything, and we shouldn't.
Actually, we have to catch the error because `getTriggerCompatibleActions` is async, but, I my opinion, since we are not doing API requests or any error-prone logic, if it rejects the promise we should just bubble up the error as any other usual component would do.

Would it make sense to add error control inside the `loadActions`* hooks? so we can forget about it in the components?

Reply:
Let's throw it!